gateway/ln-gateway/src/lightning_manager.rs

```diff
@@ -16,6 +17,10 @@ pub struct LightningManager {
 
     /// The current state of the Gateway.
     state: Arc<RwLock<GatewayState>>,
+
+    /// Task group for managing HTLC interception.
+    /// Is `Some` when the gateway is intercepting HTLCs, `None` otherwise.
+    htlc_task_group_or: Arc<RwLock<Option<TaskGroup>>>,
```

Comment:
The `Option` here is pretty ugly. Why is it necessary? We shouldn't need to recreate the `TaskGroup` anytime we disconnect from the Lightning node. Can we do something similar to what we do now, where we just pass in the `TaskGroup` as a parameter to a function, instead of storing it as a member variable here? e.g. `lightning_manager.listen_for_htlcs(task_group).await` ?

Reply:
For my clarification, are you suggesting that we change

`htlc_task_group_or: Arc<RwLock<Option<TaskGroup>>>`

to

`htlc_task_group: Arc<RwLock<TaskGroup>>`

and then add a `TaskGroup` function parameter to `LightningManager::new_initializing`?

Or are you suggesting we no longer have `LightingManager` own the HTLC task group?

Reply:
I was hoping to not have `LightningManager` own the task group. Just pass it in as a parameter to the HTLC interception function.